bokeh/application/application.py

```diff
@@ -100,9 +101,23 @@ class Application(object):
     # will function without bringing in tornado.
     _is_a_bokeh_application_class = True
 
-    def __init__(self, *handlers):
+    def __init__(self, *handlers, metadata=None):
```

Comment:
Kwargs-only args do not wok with Python2, we will need to find another way to express this, perhaps a Python getter/setter for a `metadata`property